trieste/acquisition/function.py

```diff
@@ -264,6 +284,21 @@ def __init__(self, search_space: SearchSpace, num_samples: int = 10, grid_size:
             raise ValueError(f"grid_size must be positive, got {grid_size}")
         self._grid_size = grid_size
 
+        if type(num_fourier_features) is int:
```

Comment:
It's more Pythonic to check for optional values like this:

```suggestion
        if num_fourier_features is not None:
```